chapters/en/chapter6/pre-trained_models.mdx

```diff
@@ -0,0 +1,199 @@
+
+After pre-training, the entire encoder-decoder backbone is fine-tuned for each individual task. At this step, only the 
+pre-nets and post-nets relevant to the specific task are employed. For example, to use SpeechT5 for text-to-speech, you'd 
+need the text encoder pre-net for the text inputs and the speech decoder pre- and post-nets for the speech outputs. 
+
+This approach allows to obtain several models fine-tuned for different speech tasks that all benefit from the initial 
+pre-training on unlabeled data.  
+
+<Tip> 
+	Even though the fine-tuned models start out using the same set of weights from the shared pre-trained model, the 
+final versions are all quite different in the end. You can't take a fine-tuned ASR model and swap out the pre-nets and 
+post-net to get a working TTS model, for example. SpeechT5 is flexible, but not that flexible ;)
+
+</Tip>
+
+Let's see what are the pre- and post-nets that SpeechT5 uses for the TTS task specifically:
+
+* Text encoder pre-net: A text embedding layer that maps text tokens to the hidden representations that the encoder expects. This is similar to what happens in an NLP model such as BERT.
+* Speech decoder pre-net: This takes a log mel spectrogram as input and uses a sequence of linear layers to compress the spectrogram into hidden representations. 
+* Speech decoder post-net: This predicts a residual to add to the output spectrogram and is used to refine the results.
+
+When combined, this is what SpeechT5 architecture for text-to-speech looks like:
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/speecht5/tts.jpg" alt="SpeechT5 architecture for TTS">
+</div>
+
+As you can see, the output is a log mel spectrogram and not a final waveform. If you recall, we briefly touched on 
+this topic in [Unit 3](../chapter3/introduction#spectrogram-output). It is common for models that generate audio to produce 
+a log mel spectrogram, which needs to be converted to a waveform with an additional neural network known as a vocoder.
+
+Let's see how you could do that.
+
+First, let's load the fine-tuned TTS SpeechT5 model from the 🤗 Hub, along with the processor object used for tokenization 
+and feature extraction:
+
+```python
+from transformers import SpeechT5Processor, SpeechT5ForTextToSpeech
+
+processor = SpeechT5Processor.from_pretrained("microsoft/speecht5_tts")
+model = SpeechT5ForTextToSpeech.from_pretrained("microsoft/speecht5_tts")
+```
+
+Next, tokenize the input text.
+
+```python
+inputs = processor(text="Don't count the days, make the days count.", return_tensors="pt")
+```
+
+The SpeechT5 TTS model is not limited to creating speech for a single speaker. Instead, it uses so-called speaker embeddings 
+that capture a particular speaker's voice characteristics. 
+
+[//]: # (TODO: We should elaborate here how the speaker embeddings are obtained, what they represent, etc.)
```

Comment:
+1 - this will become more important for future models such as Bark which are currently being integrated

Reply:
@sanchit-gandhi Would you like to add a text snippet on this?

Reply:
Or provide me with some pointers to speed up my own research?

Reply:
Some useful links:

1. Short overview: https://github.com/RF5/simple-speaker-embedding#what
2. More descriptive: https://www.compriseh2020.eu/usage-of-speaker-embeddings-for-more-inclusive-speech-recognition/ (see section "the role of speaker embeddings" -> we care about x-vectors)
3. The x-vector paper: https://www.danielpovey.com/files/2018_icassp_xvectors.pdf